src/main/java/io/jenkins/plugins/opentelemetry/OpenTelemetryConfiguration.java

```diff
@@ -43,17 +50,34 @@ public String getTrustedCertificatesPem() {
         return trustedCertificatesPem;
     }
 
+    @Nonnull
+    public int getCollectorTimeout() {
```

Comment:
I'm not clear on what "Collector" means, I see a risk of misunderstanding with the OpenTelemetry Collector.

Is it correct that it's the same as `OTEL_EXPORTER_OTLP_TIMEOUT ` and is used for `io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder#setTimeout(long, java.util.concurrent.TimeUnit)` and `io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporterBuilder#setTimeout(long, java.util.concurrent.TimeUnit)`?

Would a name like "exporterTimeoutMillis" make sense?

Reply:
Exactly, `span` and `metrics` use the same timeout:
- https://github.com/open-telemetry/opentelemetry-java/blob/43f1ecf7ede49104c2aac1c9e5448b7cf7859ce7/exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java#L51-L60
- https://github.com/open-telemetry/opentelemetry-java/blob/3528123c3512e93ce34b7929039158aeaab5ac08/exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporterBuilder.java#L47-L56

Since both exporters use the same value, I decided to use the name defined in the javadoc:

> Sets the maximum time to wait for the collector to process an exported batch of metrics

Somehow the class is an `exporter` but the docs refers to the `collector`.  I don't have any strong opinion, regarding the `Millis` suffix, I wanted to move away from the time `unit` in the name, but nothing but for some cosmetic reasons.

Reply:
See https://github.com/jenkinsci/opentelemetry-plugin/pull/39/commits/dbb64707e6ce0db75a53e52c28f27ea759c566e4